src/rt/cpp/when.h

```diff
@@ -14,13 +14,102 @@ namespace verona::cpp
 {
   using namespace verona::rt;
 
+  template<typename T>
+  struct cown_ptr_span
+  {
+    cown_ptr<T>* array;
+    size_t length;
+
+    void constr_helper(cown_ptr<T>* arr)
+    {
+      array = reinterpret_cast<cown_ptr<T>*>(
+        snmalloc::ThreadAlloc::get().alloc(length * sizeof(cown_ptr<T>*)));
+      memset(array, 0, length * sizeof(cown_ptr<T>*));
+
+      for (size_t i = 0; i < length; i++)
+        array[i] = arr[i];
+    }
+
+    template<bool should_move = false>
+    cown_ptr_span(cown_ptr<T>* array_, size_t length_) : length(length_)
+    {
+      if constexpr (should_move == false)
+      {
+        constr_helper(array_);
+      }
+    }
+
+    cown_ptr_span(const cown_ptr_span& o)
+    {
+      length = o.length;
+      constr_helper(o.array);
+    }
+
+    ~cown_ptr_span()
+    {
+      if (array)
+      {
+        for (size_t i = 0; i < length; i++)
+          array[i].~cown_ptr<T>();
+
+        snmalloc::ThreadAlloc::get().dealloc(array);
+      }
+    }
+
+    cown_ptr_span(cown_ptr_span&& old) = delete;
+    cown_ptr_span& operator=(cown_ptr_span&&) = delete;
+    cown_ptr_span& operator=(const cown_ptr_span&) = delete;
+  };

```

Comment:
Why are these operators deleted? Could you add a comment to explain.

Reply:
There's no need to be deleted. They are not used in the current codebase. So, I marked them as deleted to avoid bugs in auto generated. I'll add a comment to clarify.